ignite/metrics/rouge.py

```diff
@@ -0,0 +1,168 @@
+from collections import defaultdict
+from typing import Callable, DefaultDict, List, Tuple, Union
+
+import torch
+
+from ignite.exceptions import NotComputableError
+from ignite.metrics import Metric
+
+# These decorators helps with distributed settings
+from ignite.metrics.metric import reinit__is_reduced, sync_all_reduce
+
+
+def _ngramify(tokens: List[str], n: int) -> DefaultDict:
+    ngrams: DefaultDict = defaultdict(int)
+    for ngram in (tuple(tokens[i : i + n]) for i in range(len(tokens) - n + 1)):
```

Comment:
```suggestion
    for ngram in (tuple(tokens[i: i + n]) for i in range(len(tokens) - n + 1)):
```
PEP

Reply:
Black can also "fix" spaces like that for indexing...

Reply:
VScode pep8 formatter does not seem to fix it on its own. I will look into it.

Reply:
Well, let's keep it like that without any change. For instance, we do not configure black and just use it as is...